netmiko/cisco/cisco_vxr_ssh.py

```diff
@@ -0,0 +1,309 @@
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import re
+import time
+import logging
+import os
+import paramiko
+
+from netmiko.cisco.cisco_xr import CiscoXrSSH
+from netmiko.netmiko_globals import BACKSPACE_CHAR
+from netmiko.utilities import get_structured_data
+
+work_dir = os.getenv('CAFYKIT_WORK_DIR')
+if work_dir:
+    formatter = logging.Formatter(fmt='%(asctime)s %(levelname)-8s %(message)s',
+                                  datefmt='%Y-%m-%d %H:%M:%S')
+    paramiko_log_folder = os.path.join(work_dir, 'test_paramiko.log')
+    paramiko.util.log_to_file(paramiko_log_folder, level="DEBUG")
+    fh = logging.FileHandler(paramiko_log_folder)
+    fh.setFormatter(formatter)
+    log = logging.getLogger("netmiko")
+    log.addHandler(fh)
+    log.setLevel(logging.DEBUG)
+    log.addHandler(fh)
+else:
+    from netmiko import log
+
+
+class CiscoVxrSSH(CiscoXrSSH):
+    """
+    CiscoVxrSSH is based of CiscoXrSSH -- CiscoBaseConnection
+    """
+
+    def __init__(self,
+                 **kwargs):
+        """Constructor
+        """
+        # 30 minutes
+        self.max_read_timeout = kwargs.get('max_read_timeout', 1800)
+        super().__init__(**kwargs)
+
+    def find_prompt(self, delay_factor=1):
+        """Finds the current network device prompt, last line only.
+
+        :param delay_factor: See __init__: global_delay_factor
+        :type delay_factor: int
+        """
+        loop_delay = 0.1
+        self.clear_buffer()
+        self.write_channel(self.RETURN)
+        time.sleep(loop_delay)
+
+        vxr_pattern = "last login"
+        autocommand_pattern = "executing autocommand"
+        cxr_pattern = "last switch-over"
+
+        prompt = None
+        start_time = time.time()
+        current_time = time.time()
+        while current_time - start_time < self.max_read_timeout and self.is_alive():
```

Comment:
If `current_time`. is replaced by `time.time()` within the loop condition, we'll not have to initialize or update it at the end of the loop.

Reply:
its used in logs as well at line 67 and 77. current_time variable is used to avoid calculating time.time() again and again